index.js

```diff
@@ -1,30 +1,42 @@
 import fp from 'fastify-plugin'
 import onFinished from 'on-finished'
-import defaults from 'defaults'
 
 import { DEFAULT_OPTIONS, HEADERS } from './lib/constants.js'
 import { Store } from './lib/store.js'
 import { convertToMs, calculateDelay } from './lib/helpers.js'
 
 const slowDownPlugin = async (fastify, settings) => {
-  const options = defaults(settings, DEFAULT_OPTIONS)
+  const options = { ...DEFAULT_OPTIONS, ...settings }
   const store = new Store(convertToMs(options.timeWindow))
 
+  if (!fastify.hasRequestDecorator('slowDown')) {
+    fastify.decorateRequest('slowDown', undefined)
+  }
+
   fastify.addHook('onClose', () => store.close())
 
   fastify.addHook('onRequest', async (req, reply) => {
     const requestCounter = store.incrementOnKey(options.keyGenerator(req))
     const delayMs = calculateDelay(requestCounter, options)
+    const hasDelay = delayMs > 0
+    const remainingRequests = Math.max(options.delayAfter - requestCounter, 0)
 
     reply.header(HEADERS.limit, options.delayAfter)
-    reply.header(
-      HEADERS.remaining,
-      Math.max(options.delayAfter - requestCounter, 0)
-    )
+    reply.header(HEADERS.remaining, remainingRequests)
+
+    if (fastify.hasRequestDecorator('slowDown')) {
+      req.slowDown = {
```

Comment:
why don't we use the same as the express plugin instead of coming up with a new structure of this decorator? https://github.com/nfriedly/express-slow-down#reqslowdown